### PR TITLE
Refactor a unittest in test_median.py

### PR DESCRIPTION
**Pull request by keisuke-umezawa**
## Motivation

`test_medain.py` can be simpler using should_prune and the ask-and-tell interface.

## Description of the changes

Use should_prune and the ask-and-tell interface.

### CODE REVIEW

File: tests/pruners_tests/test_median.py [hunk 1]
```diff
@@ -26,6 +26,7 @@ def test_median_pruner_intermediate_values(direction_value: Tuple[str, float]) -
     study = optuna.study.create_study(direction=direction)
 
     trial = optuna.trial.Trial(study, study._storage.create_new_trial(study._study_id))
+    trial = study.ask()
     trial.report(1, 1)
     study._storage.set_trial_state(trial._trial_id, TrialState.COMPLETE)
```
Comment (hvy) [inline tests/pruners_tests/test_median.py:31]: Is using tell outside the scope of this PR?

File: tests/pruners_tests/test_median.py [hunk 2]
```diff
@@ -26,6 +26,7 @@ def test_median_pruner_intermediate_values(direction_value: Tuple[str, float]) -
     study = optuna.study.create_study(direction=direction)
 
     trial = optuna.trial.Trial(study, study._storage.create_new_trial(study._study_id))
```
Comment (hvy) [inline tests/pruners_tests/test_median.py:28]: This line is no longer needed because of the addition on line 29.
```suggestion
```

**Review by hvy — COMMENTED** [review]
Thanks for the PR. The motivation sounds good. How about applying the same fixes to other test cases in `test_median.py`?

**Comment by keisuke-umezawa** [comment]
@hvy Thank you for reviewing. I followed your comment.

> Thanks for the PR. The motivation sounds good. How about applying the same fixes to other test cases in test_median.py?

File: tests/pruners_tests/test_median.py [hunk 3]
```diff
@@ -9,97 +9,97 @@
...
+    study = optuna.study.create_study(pruner=pruner)
 
-    trial = optuna.trial.Trial(study, study._storage.create_new_trial(study._study_id))
+    trial = study.ask()
     trial.report(float("nan"), 1)
     # A pruner is not activated if the study does not have any previous trials.
-    assert not pruner.prune(study=study, trial=study._storage.get_trial(trial._trial_id))
-    study._storage.set_trial_state(trial._trial_id, TrialState.COMPLETE)
+    assert not trial.should_prune()
+    study.tell(trial, -1)
```
Comment (keisuke-umezawa) [inline tests/pruners_tests/test_median.py:50]: What would be a good value here? `study.tell(trial, float("nan"))` rises ValueError.

Comment (hvy) [reply tests/pruners_tests/test_median.py:49]: In my understanding, the value doesn't matter for the purpose of these unit tests so I don't have a strong opinion but something that communicates this, like -1 (or 0) sounds reasonable to me. We could also leave a comment.

**Comment by hvy** [comment]
@himkt could you also take a look at this PR?

File: tests/pruners_tests/test_median.py (hunk 3 above)
Comment (himkt) [reply tests/pruners_tests/test_median.py:49]: It also makes sense for me to use `-1` here.

**Comment by hvy** [comment]
Thanks for the updated @keisuke-umezawa, LGTM!

**Pull request merged**